tests/robot-tests/reports.py

```diff
@@ -0,0 +1,109 @@
+        
+        for file in glob.glob(rf"{test_run_foldername}{os.sep}*screenshot*"):
+            _copy_to_destination_folder(test_run_foldername, file.split(os.sep)[-1], main_results_folder)
+
+        for file in glob.glob(rf"{test_run_foldername}{os.sep}*capture*"):
+            _copy_to_destination_folder(test_run_foldername, file.split(os.sep)[-1], main_results_folder)
+
+
+def log_report_results(number_of_test_runs: int, failing_suites: []):
+    logger.info(f"Log available at: file://{os.getcwd()}{os.sep}{main_results_folder}{os.sep}log.html")
+    logger.info(f"Report available at: file://{os.getcwd()}{os.sep}{main_results_folder}{os.sep}report.html")
+    logger.info(f"Number of test runs: {number_of_test_runs}")
+
+    if failing_suites:
+        logger.info(f"Number of failing suites: {len(failing_suites)}")
+        logger.info(f"Failing suites:")
+        [logger.info(r"  * file://" + suite) for suite in failing_suites]
+    else:
+        logger.info("\nAll tests passed!")
+
+
+def create_report_from_output_xml(test_results_folder: str):
+    create_args = [
+        "--outputdir",
+        f"{test_results_folder}/",
+        "-o",
+        "output.xml",
+        "--xunit",
+        "xunit.xml",
+        f"{test_results_folder}/output.xml"
+    ]
+    robot_rebot_cli(create_args, exit=False)
+
+
+def _merge_test_reports(test_results_folder):
+    merge_args = [
+        "--outputdir",
+        f"{main_results_folder}/",
+        "-o",
+        "output.xml",
+        "--xunit",
+        "xunit.xml",
+        "--prerebotmodifier",
+        "report-modifiers/CheckForAtLeastOnePassingRunPrerebotModifier.py",
+        "--merge",
+        f"{main_results_folder}/output.xml",
+        f"{test_results_folder}/output.xml",
+    ]
+    robot_rebot_cli(merge_args, exit=False)
+
+
+def filter_out_passing_suites_from_report_file(path_to_original_report: str, path_to_filtered_report: str):
+    
+    with open(path_to_original_report, "rb") as report:
+        
+        report_contents = report.read()
+        report = BeautifulSoup(report_contents, features="xml")
+        
+        passing_suite_ids = _get_passing_suite_ids_from_report(report)
+        
+        suite_elements = report.find_all('suite', recursive=True)
+        [suite_element.extract() for suite_element in suite_elements if suite_element.get('id') in passing_suite_ids]
+        
+        suite_stats = report.find_all('stat', recursive=True)
+        [suite_stat.extract() for suite_stat in suite_stats if suite_stat.get('id') in passing_suite_ids]
+    
+        with open(path_to_filtered_report, "a") as filtered_file:
+            filtered_file.write(report.prettify())
+
+        
+def _get_passing_suite_ids_from_report(report: BeautifulSoup) -> []: 
+    suite_results = report.find('statistics').find('suite').find_all('stat')
+    passing_suite_results = [suite_result for suite_result in suite_results if int(suite_result.get('fail')) == 0]
```

Comment:
**praise** very nice 😎

Reply:
Why thank you thank you :D  Handy to have all that pre-existing BeautifulSoup code to use as a reference of course!